### PR TITLE
Foods CRUD

### PR DESCRIPTION
**Pull request by kolyaventuri**
- Created food model
- Creates CRUD functionality for foods including sad path tests

### CODE REVIEW

File: app/controllers/api/v1/foods_controller.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    food = Food.where(id: params[:id]).first
+
+    if food.nil?
+      render nothing: true, status: 404 and return
+    end
+
+    if food.delete
+      render nothing: true, status: 204
+    else
+      render nothing: true, status: 404 and return
```
Comment (kolyaventuri) [inline app/controllers/api/v1/foods_controller.rb:47]: @corywest I feel like this is possibly redundant? As thought it is one of those "it could happen, however unlikely" errors. Not sure how to properly test this behavior, as it would have to fail at the SQL transaction point presumably (as if it was some error with the database) or am I wrong about that? Is there a decent way to test something like this? Or for the scope of this project, should I not bother digging too deep into an edge case like that.

Nearly-identical situation on like 33

Comment (corywest) [reply app/controllers/api/v1/foods_controller.rb:47]: @kolyaventuri Yes, it's a bit redundant. You could assume that a SQL transaction itself could fail at some point, but it's unlikely. I would throw that test case out for now. We can chat more about it later if you'd like!

Comment (kolyaventuri) [reply app/controllers/api/v1/foods_controller.rb:47]: Excellent, thanks! I would definitely like to talk about it later, how to handle/test it and such
